### ISSUE TITLE
Sorted column support in generate_source

### ISSUE THREAD

**Issue by abupp** [dbt-labs/dbt-codegen#157]
### Describe the feature
As a user I would like a new optional argument to the generate_source operation to sort columns.

### Describe alternatives you've considered
Currently, I generate the source YAML and then copy and paste it into a spreadsheet if I want to sort it.

### Additional context
I do not think that that feature would be database specific, though I have only tested it against Snowflake.

### Who will this benefit?
This feature be particularly  useful for tables that contain hundreds of columns, making them much easier to work with and maintain. I work for a company that manages data models related to Salesforce and Jira tables. Both systems support the creation and deletion of custom fields, and are moving targets that require regular maintenance of dbt models.

### Are you interested in contributing this feature?
Sure, here's some suggested code:

In https://github.com/dbt-labs/dbt-codegen/blob/main/macros/generate_source.sql change line 62 from:

        {% set columns=adapter.get_columns_in_relation(table_relation) %}
to:

        {% set columns=adapter.get_columns_in_relation(table_relation) | sort(attribute='column') %}

This could be done conditionally if a the macro accepted an additional "sort_columns" argument (that defaulted to False):

        {% if sort_columns %}
            {% set columns=adapter.get_columns_in_relation(table_relation) | sort(attribute='column') %}
        {% else %}
            {% set columns=adapter.get_columns_in_relation(table_relation) %}
        {% endif %}

**Comment by dave-connors-3** [comment]
hey @abupp ! thanks for the issue -- i think this would be a great enhancement. Can you think of a scenario where someone would _not_ want sorted columns? trying to decide if this should be the default or configurable

**Comment by abupp** [comment]
Greetings @dave-connors-3 You're quite welcome. Yes, I think there are likely to be scenarios where sorting would not be preferable, but when dealing w/ tables that have large column sets, I would not likely agree with the rationale.  I have implemented sorting in our own dbt project, mostly it was not hard to do, and I found the branch test setup instructions were not very good for the Windows system that I run on. I could send you the code. I also adding sorting to the generate_base_model macro.
